lib/typing/columns/columns.go

```diff
@@ -310,3 +271,32 @@ func ColumnsUpdateQuery(columns []string, columnsToTypes Columns, destKind const
 
 	return strings.Join(_columns, ",")
 }
+
+func processToastStructCol(colName string, destKind constants.DestinationKind) string {
+	if destKind == constants.BigQuery {
+		return fmt.Sprintf(`%s= CASE WHEN COALESCE(TO_JSON_STRING(cc.%s) != '{"key":"%s"}', true) THEN cc.%s ELSE c.%s END`,
+			colName, colName, constants.ToastUnavailableValuePlaceholder,
+			colName, colName)
+	} else if destKind == constants.Redshift {
+		return fmt.Sprintf(`%s= CASE WHEN COALESCE(cc.%s != JSON_PARSE('{"key":"%s"}'), true) THEN cc.%s ELSE c.%s END`,
+			colName, colName, constants.ToastUnavailableValuePlaceholder, colName, colName)
+	} else if destKind == constants.MSSQL {
+		// Microsoft SQL Server doesn't allow boolean expressions to be in the COALESCE statement.
+		return fmt.Sprintf("%s= CASE WHEN COALESCE(cc.%s, {}) != {'key': '%s'} THEN cc.%s ELSE c.%s END",
+			colName, colName, constants.ToastUnavailableValuePlaceholder, colName, colName)
+	} else {
```

Comment:
Is this case just Snowflake? If so it might be better explicitly check that it's Snowflake and then error if it isn't so that when we add a new DWH type we remember to implement this for it.

Reply:
It is, I'll leave a TODO, erroring out will require changing the function signature